File: 4 - Class vs. Instance/Solution.py
class Person:
    def __init__(self, initialAge):
        if initialAge > 0:
            self.age = initialAge
        else:
            print("Age is not valid, setting age to 0.")
            self.age = 0

    def amIOld(self):
        if self.age < 13:
            print("You are young.")
        elif self.age < 18:
            print("You are a teenager.")
        else:
            print("You are old.")

    def yearPasses(self):
        self.age += 1


t = int(input())
for i in range(0, t):
    age = int(input())
    p = Person(age)
    p.amIOld()
    for j in range(0, 3):
        p.yearPasses()
    p.amIOld()
    print("")

File: 4 - Class vs. Instance/Solution.kt
class Person(initialAge: Int) {
    var age = 0

    init {
        age = if (initialAge > 0) initialAge
        else {
            println("Age is not valid, setting age to 0.")
            0
        }
    }

    fun amIOld() {
        when {
            age < 13 -> println("You are young.")
            age < 18 -> println("You are a teenager.")
            else -> println("You are old.")
        }
    }

    fun yearPasses() = age++
}

fun main(args: Array<String>) {
    val T = readLine()!!.toInt()
    for (i in 0 until T) {
        val age = readLine()!!.toInt()
        val p = Person(age)
        p.amIOld()
        for (j in 0..2) {
            p.yearPasses()
        }
        p.amIOld()
        println()
    }
}
